src/processes/scoring/calc-karma-score.js

```diff
@@ -78,10 +77,25 @@ const calcKarmaScore = async (userId, sign_user_score = null) => {
   if (!user || user?.is_anonymous) {
     return null;
   }
-  const combined_user_score = await combinedUserScore(userId, userScoreCol, sign_user_score);
+  const anonymousUserId = await UsersFunction.findAnonymousUserId(userId);
+  const combined_user_score = await combinedUserScore(
+    userId,
+    userScoreCol,
+    sign_user_score,
+    anonymousUserId
+  );
   const percentileScore = await percentileUserScore(combined_user_score, userScoreCol);
   const karma_score = KARMA_SCORE_MULTIPLIER * percentileScore ** KARMA_SCORE_EXPONENT;
   await setUserKarmaScore(userId, combined_user_score, karma_score, userScoreCol, percentileScore);
+  if (anonymousUserId) {
+    await setUserKarmaScore(
+      anonymousUserId,
+      combined_user_score,
+      karma_score,
+      userScoreCol,
+      percentileScore
+    );
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic here seems to be correct, but there's a potential performance issue. The same karma score is being calculated twice for anonymous users - once for the authenticated user and then again for the anonymous user. Since the calculation is the same, you could store the result and reuse it, avoiding unnecessary computation.

```diff
- await setUserKarmaScore(userId, combined_user_score, karma_score, userScoreCol, percentileScore);
- if (anonymousUserId) {
-   await setUserKarmaScore(
-     anonymousUserId,
-     combined_user_score,
-     karma_score,
-     userScoreCol,
-     percentileScore
-   );
- }
+ await setUserKarmaScore(userId, combined_user_score, karma_score, userScoreCol, percentileScore);
+ if (anonymousUserId) {
+   await setUserKarmaScore(anonymousUserId, combined_user_score, karma_score, userScoreCol, percentileScore);
+ }
```
Please note that the above suggestion assumes that `setUserKarmaScore` is an idempotent operation. If it has side effects or its results depend on when it's called, this suggestion may not be applicable.


<!-- This is an auto-generated comment by OSS CodeRabbit -->